pkg/util/k8sutil_test.go

```diff
@@ -0,0 +1,153 @@
+						},
+					}},
+			}
+			testpod1 := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: "default"}, Spec: v1.PodSpec{Containers: []v1.Container{{Image: "testimage"}}}}
+			result = IsPodReady(testpod)
+			result1 = IsPodReady(testpod1)
+		})
+		It("pod ready should be true", func() {
+			Ω(result).To(Equal(true))
+		})
+		It("pod ready should be false", func() {
+			Ω(result1).To(Equal(false))
+		})
+	})
+	Context("podFaulty", func() {
+		var result, result1 bool
+		BeforeEach(func() {
+			testpod := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: "default", Name: "test"}, Spec: v1.PodSpec{Containers: []v1.Container{{Image: "testimage"}}},
+				Status: v1.PodStatus{
+					ContainerStatuses: []v1.ContainerStatus{
+						{
+							Name: "test",
+							State: v1.ContainerState{
+								Waiting: &v1.ContainerStateWaiting{
+									Reason: "CrashLoopBackOff",
+								},
+							},
+						},
+					}},
+			}
+			testpod1 := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: "default", Name: "test"}, Spec: v1.PodSpec{Containers: []v1.Container{{Image: "testimage"}}},
+				Status: v1.PodStatus{
+					ContainerStatuses: []v1.ContainerStatus{
+						{
+							Name:  "test",
+							State: v1.ContainerState{},
+						},
+					}},
+			}
+			result, _ = IsPodFaulty(testpod)
+			result1, _ = IsPodFaulty(testpod1)
+		})
+		It("pod faulty should be true", func() {
+			Ω(result).To(Equal(true))
+		})
+		It("pod faulty should be false", func() {
+			Ω(result1).To(Equal(false))
+		})
+	})
+	Context("NewEvent", func() {
+		var bk *v1alpha1.BookkeeperCluster
+		var event *corev1.Event
+		BeforeEach(func() {
+			bk = &v1alpha1.BookkeeperCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			bk.WithDefaults()
+			message := "upgrade failed"
+			event = NewEvent("bookie", bk, "UPGRADE_ERROR", message, "Error")
+		})
+
+		It("Event size should not be zero", func() {
+			Ω(event.Size()).ShouldNot(Equal(0))
+		})
+	})
+	Context("NewApplicationEvent", func() {
+		var bk *v1alpha1.BookkeeperCluster
+		var event *corev1.Event
+		BeforeEach(func() {
+			bk = &v1alpha1.BookkeeperCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			bk.WithDefaults()
+			message := "upgrade failed"
+			event = NewApplicationEvent("bookie", bk, "UPGRADE_ERROR", message, "Error")
+		})
+		It("Event size should not be zero", func() {
+			Ω(event.Size()).ShouldNot(Equal(0))
+		})
+	})
+	Context("WaitForClusterToTerminate", func() {
+		var bk, bk1 *v1alpha1.BookkeeperCluster
+		var client client.Client
+		var err error
+		BeforeEach(func() {
+
+			bk = &v1alpha1.BookkeeperCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			bk.WithDefaults()
+			s := scheme.Scheme
+			s.AddKnownTypes(v1alpha1.SchemeGroupVersion, bk)
+			s.AddKnownTypes(v1alpha1.SchemeGroupVersion, bk1)
+			client = fake.NewFakeClient(bk)
+			err = WaitForClusterToTerminate(client, bk)
+		})
+		It("should  be nil", func() {
```

Comment:
Could we try to get the bk cluster and check whether it has actually terminated here?

Reply:
This is not feasible, even if we create cluster with some pods, we cant terminate the pods while the function is running